### PR TITLE
Introduce tslint

### PR DESCRIPTION
**Pull request by jedmao**

### CODE REVIEW

File: src/commands/generateEditorConfig.ts [hunk 1]
```diff
@@ -1,16 +1,27 @@
...
 import * as path from 'path';
-import {workspace, window} from 'vscode';
-import {Utils} from '../utils';
+import {
+	workspace,
+	window
+} from 'vscode';
+import {
+	Utils
+} from '../utils';
```
Comment (SamVerschueren) [inline src/commands/generateEditorConfig.ts]: We could use `export default class Utils` and then use `import Utils from '../utils';`.

File: src/editorConfigMain.ts [hunk 2]
```diff
@@ -1,6 +1,13 @@
-import {ExtensionContext, commands} from 'vscode';
-import {DocumentWatcher} from './documentWatcher';
-import {generateEditorConfig} from './commands/generateEditorConfig';
+import {
+	ExtensionContext,
+	commands
+} from 'vscode';
+import {
+	DocumentWatcher
+} from './documentWatcher';
```
Comment (SamVerschueren) [inline src/editorConfigMain.ts]: `export default class DocumentWatcher`

`import DocumentWatcher from './documentWatcher';`

File: src/utils.ts [hunk 3]
```diff
@@ -1,16 +1,29 @@
...
 export class Utils {
 
 	/**
 	 * Convert .editorconfig values to vscode editor options
 	 */
-	public static fromEditorConfig(config: editorconfig.knownProps, defaults: EditorSettings): any {
+	public static fromEditorConfig(
+		config: editorconfig.knownProps,
+		defaults: EditorSettings
+	) {
```
Comment (SamVerschueren) [inline src/utils.ts]: Actually I like the old style more to be honest. But I see the benefit for diffs. It's easier to see what changed if one type changes or an extra parameter was added. The code now looks a little bit cluttered to me.

File: src/commands/generateEditorConfig.ts (hunk 1 above)
Comment (jedmao) [inline src/commands/generateEditorConfig.ts]: doin' it now

Comment (SamVerschueren) [inline src/commands/generateEditorConfig.ts]: :+1:

File: src/documentWatcher.ts [hunk 4]
```diff
@@ -1,14 +1,33 @@
...
+} from './utils';
+import {
+	transform as trimTrailingWhitespaceTransform
+} from './transformations/trimTrailingWhitespace';
+import {
+	transform as insertFinalNewlineTransform
+} from './transformations/insertFinalNewline';
+import {
+	IEditorConfigProvider
+} from './interfaces/editorConfigProvider';
```
Comment (SamVerschueren) [inline src/documentWatcher.ts]: `export default IEditorConfigProvider`

File: src/documentWatcher.ts [hunk 5]
```diff
@@ -1,14 +1,33 @@
...
+import {
+	EditorSettings
+} from './interfaces/editorSettings';
```
Comment (SamVerschueren) [inline src/documentWatcher.ts]: `export default EditorSettings`

**Pull request merged**